SQF/dayz_code/actions/jerry_fill.sqf

```diff
@@ -11,22 +11,22 @@ _magazines = magazines player;
 _fuelAmount = _cursorTarget getVariable "FuelAmount";
 
 if (isNil "_fuelAmount") then {
-	_fuelAmount = floor(Random dayz_randomMaxFuelAmount);
+	_fuelAmount = floor(Random dayz_randomMaxFuelAmount) max (dayz_randomMaxFuelAmount * 0.10);
     _cursorTarget setVariable ["FuelAmount",_fuelAmount,true];
 };
 
-if (_fuelAmount < 5) exitWith { format[localize "str_fill_notenough",typeOf _cursorTarget,_fuelAmount] call dayz_rollingMessages; };
+_qty5 = {_x == "ItemFuelcanEmpty"} count _magazines;
+_qty20 = {_x == "ItemJerrycanEmpty"} count _magazines;
+_qty210 = {_x == "ItemFuelBarrelEmpty"} count _magazines;
 
-diag_log format["Fill Jerry, %1 - %2",_cursorTarget,_fuelAmount];
+_fuelNeeded = (_qty5 * 5) + (_qty20 * 20) + (_qty210 * 210);
 
-_fuelcans = ["ItemFuelcanEmpty","ItemJerrycanEmpty","ItemFuelBarrelEmpty"];
+if (_fuelAmount < _fuelNeeded) exitWith {dayz_actionInProgress = false; format[localize "str_fill_notenough",typeOf _cursorTarget,_fuelAmount,_fuelNeeded] call dayz_rollingMessages;};
```

Comment:
This is always going to exit if there isn't enough fuel to fill every single can in your gear.

It is better to leave it so it fills the cans that it has enough for and leaves the rest empty. Otherwise you will have to move some cans out of your inventory or drop them every time there isn't enough fuel to fill all of them at once.